api/src/main/java/org/commonjava/maven/galley/TransferManager.java

```diff
@@ -75,16 +75,16 @@ boolean delete( ConcreteResource resource )
     boolean delete( ConcreteResource resource , EventMetadata eventMetadata  )
         throws TransferException;
 
-    boolean publish( ConcreteResource resource, InputStream stream, long length )
+    boolean publish( ConcreteResource resource, InputStream stream, long length, EventMetadata metadata )
```

Comment:
Ideally these should have the old method signatures preserved, where they call the new one with a `new EventMetadata()` to default over to Maven package type.

I remember, when I introduced EventMetadata in the first place, that adding these original methods back in made it harder to find consistency problems in the calls I made from Indy into TransferManager...we can leave it until later to restore backward compat if you like...as long as it happens before the next galley release, I think we should be okay.